### PR TITLE
Custom DC: Standalone Terraform module for BT automation

### PR DESCRIPTION
**Pull request by Fructokinase**
This standalone module can be called directly to automate BT and GCF setup. This PR is copied over from datacommonsorg/website#1895, but no longer needs to curl for source files.

This PR only tests whether the resources can be deployed. After #165 is submitted, follow up PRs will change BT manual scaling to autoscaling (BT created from this script will have autoscale on).

BT created.
https://pantheon.corp.google.com/bigtable/instances/dc-graph-alex20221219/overview?mods=-monitoring_api_staging&project=datcom-website-dev

GCF created.
https://pantheon.corp.google.com/functions/details/us-central1/prophet-cache-trigger-datcom-website-dev-alex20221219?env=gen1&mods=-monitoring_api_staging&project=datcom-website-dev

### CODE REVIEW

File: bigtable_automation/terraform/main.tf [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+resource "google_cloudfunctions_function" "bt_automation" {
+  name        = format(
+      "prophet-cache-trigger-%s%s", var.project_id, local.resource_suffix)
+  project        = var.project_id
+  description = "For triggering BT cache build on gcs file writes."
+  runtime     = "go116"
+  region      = var.region
+
+  timeout                      = 300
+  entry_point                  = "PrivateBTImportController"
```
Comment (shifucun) [inline bigtable_automation/terraform/main.tf:68]: Can change this to CustomBTImportController now..

File: bigtable_automation/terraform/variables.tf [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+variable "dc_resource_bucket" {
+  type        = string
+  description = "Name of the GCS bucket that stores DC resources. Do not include the scheme(gs://)."
+}
+
+variable "csv2bt_template_path" {
+  type        = string
+  description = "Full GCS path, including scheme, to the Dataflow template to load BT cache."
+  default     = "gs://datcom-templates/templates/flex/csv_to_bt.json"
```
Comment (shifucun) [inline bigtable_automation/terraform/variables.tf:41]: Is this the old or new path?

File: bigtable_automation/terraform/main.tf (hunk 1 above)
Comment (Fructokinase) [reply bigtable_automation/terraform/main.tf:68]: Done.

File: bigtable_automation/terraform/variables.tf (hunk 2 above)
Comment (Fructokinase) [reply bigtable_automation/terraform/variables.tf:42]: This is the new path. Added a comment.

**Review by shifucun — APPROVED** [review]

**Pull request merged**